src/raspi/component/can/can_bus.py

```diff
@@ -1,64 +1,64 @@
 import can
-import cantools
-import math
-import time
 import logging
+from typing import Callable
+from dataclasses import dataclass
 
+from state_management import (
+    create_generic_context,
+    device,
+    device_action,
+    device_parser,
+)
+
+@device
+@dataclass(slots = True)
 class CanBus:
     bus: can.Bus
-    db: object
+    listeners: dict
+    notifier : can.Notifier
 
     def __init__(self,channel = "can0",interface = "socketcan",config_context = None, ignore_config = False, **kwargs):
-        self.db = cantools.database.load_file("src/raspi/odrive-cansimple.dbc")
         self.bus = can.Bus(channel,interface,config_context,ignore_config,**kwargs)
-    
-
-# TODO: Move MotorController into it's own file.
-class MotorController:
-    axis_id: int
-    can: CanBus
+        self.listeners = {}
+        self.notifier = can.Notifier(self.bus, [lambda msg: read_message(self, msg)])
 
+def read_message(can_bus: CanBus, msg) -> None:
+    axisId = msg.arbitration_id >> 5
+    if axisId in can_bus.listeners:
+        can_bus.listeners[axisId](msg)
+    else:
+        logging.info("Unhandled message with id " + hex(msg.arbitration_id))
```

Comment:
You can log info regardless of handle or not, and log warn if it was not handled.